src/main/scala/fs2/async/AsyncExt.scala

```diff
@@ -0,0 +1,208 @@
+package fs2.async
+
+import java.util.concurrent.atomic.{AtomicBoolean, AtomicReference}
+
+import fs2.internal.Actor
+import fs2.util.Task.{Result, Callback}
+import fs2.{Strategy, Async}
+import fs2.util.{Task, Free}
+
+import scala.collection.immutable.Queue
+import scala.util.control.NonFatal
+
+
+/**
+ * Created by pach on 24/10/15.
+ */
+trait AsyncExt[F[_]] extends Async[F] {
+
+  /**
+   * Modifies `ref` by applying pure function `f` to create computation `F` that's value
+   * will be used to modify the `ref`, once computed.
+   *
+   * During evaluation of `F` all other modifications will be queued up,
+   * but the `get` will receive the `A` that was supplied to `f`.
+   *
+   * If the `ref` is not yet set, this queues up until `ref` is set for the first time.
+   *
+   * Note that order of queued modifications is not preserved
+   *
+   * Likewise with `set` Resulting task may complete before final value is applied to `ref`,
+   * but it is guaranteed that it will be applied before any other modify/set on the `ref`.
+   *
+   */
+  def modify[A](ref:Ref[A])(f: A => F[A]):F[(A,A)]
+
+  /**
+   * Like `modify` but allows to interrupt the modify task (Interruptee)
+   * in case the task is waiting for the `ref` to be set for the first time by evaluating the
+   * second returned task (Interruptor)
+   *
+   * Interruptor is guaranteed to complete always with true, if Interruptee was interrupted before
+   * evaluation started, and false if Interuptee was completed before interruption.
+   *
+   * Note that Interruptor won't be able to interrupt computation after `f` was evaluated and produced computation.
+   */
+  def cancellableModify[A](r:Ref[A])(f:A => F[A]):F[(F[(A,A)], F[Boolean])]
```

Comment:
is the `Boolean` needed? This signature is already pretty complicated. When would I care whether or not my cancellation cancelled an already completed task or one that was ongoing? I feel like if I really needed that info I could track it in some other way.